app/wizards/schools/ects/teacher_leaving_wizard/check_answers_step.rb

```diff
@@ -0,0 +1,42 @@
+module Schools
+  module ECTs
+    module TeacherLeavingWizard
+      class CheckAnswersStep < ECTs::Step
+        attr_accessor :leaving_on
+
+        def previous_step = :edit
+
+        def next_step = :confirmation
+
+        def leaving_date
+          return if leaving_on.blank?
+
+          Schools::Validation::LeavingDate
+            .new(date_as_hash: leaving_on)
+            .value_as_date
+        end
+
+        def save!
+          return false unless leaving_date
```

Comment:
Can avoid the double parsing by `date = leaving_date` then `return false unless date` etc.